concent_api/core/models.py

```diff
@@ -485,3 +493,478 @@ def clean(self):
+            raise ValidationError({
+                'price': 'Price must be an integer and bigger than or equal 0'
+            })
+
+
+class StoredTaskFailure(AbstractMessageModel):
+
+    ITEMS = [
+        'task_to_compute',
+        'err',
+        'header',
+        'sig',
+    ]
+
+    if not len(ITEMS) == len(message.tasks.TaskFailure().__slots__):
+        raise ValidationError({
+            'StoredTaskFailure': 'StoredTaskFailure model has not same fields as golem_messages.message.tasks.TaskFailure()'
+        })
+
+    for model_item, golem_message_item in zip(ITEMS, message.tasks.TaskFailure().__slots__):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'StoredTaskFailure': 'StoredTaskFailure model has not same fields as golem_messages.message.tasks.TaskFailure()'
+            })
+
+    task_to_compute = ForeignKey(StoredTaskToCompute)
+    err = TextField()
+
+    '''
+    VALIDATION FOR err
+    '''
+
+
+class StoredCannotComputeTask(AbstractMessageModel):
+
+    ITEMS = [
+        'task_to_compute',
+        'reason',
+        'header',
+        'sig',
+    ]
+
+    if not len(ITEMS) == len(message.tasks.CannotComputeTask().__slots__):
+        raise ValidationError({
+            'StoredCannotComputeTask': 'StoredCannotComputeTask model has not same fields as golem_messages.message.tasks.CannotComputeTask()'
+        })
+
+    for model_item, golem_message_item in zip(ITEMS, message.tasks.CannotComputeTask().__slots__):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'StoredCannotComputeTask': 'StoredCannotComputeTask model has not same fields as golem_messages.message.tasks.CannotComputeTask()'
+            })
+
+    class CannotComputeTaskReason(ChoiceEnum):
+        WrongCTD = 'WrongCTD'
+        WrongKey = 'WrongKey'
+        WrongAddress = 'WrongAddress'
+        WrongEnvironment = 'WrongEnvironment'
+        NoSourceCode = 'NoSourceCode'
+        WrongDockerImages = 'WrongDockerImages'
+        ConcentRequired = 'ConcentRequired'
+        ConcentDisabled = 'ConcentDisabled'
+
+    if not len(CannotComputeTaskReason.__members__) == len(message.tasks.CannotComputeTask.REASON.__members__):
+        raise ValidationError({
+            'CannotComputeTaskReason': 'CannotComputeTaskReason Enum has not same fields as golem_messages.message.tasks.CannotComputeTask.REASON'
+        })
+
+    for model_item, golem_message_item in zip(CannotComputeTaskReason.__members__, message.tasks.CannotComputeTask.REASON.__members__):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'CannotComputeTaskReason': 'CannotComputeTaskReason Enum has not same fields as golem_messages.message.tasks.CannotComputeTask.REASON'
+            })
+
+    task_to_compute = ForeignKey(StoredTaskToCompute)
+    reason = CharField(max_length=32, choices=CannotComputeTaskReason.choices())
```

Comment:
Why 32? You should create a function that computes the maximum possible length for this enum and call it here.

Reply:
But it will require new migration each time a new entry is added to enum extending current max value. We already discuss it, it can't be done this way.